test/cli/test_patch.py

```diff
@@ -0,0 +1,212 @@
+                        time_periods=1,
+                        drop_bounds=True,
+                        variables=dict(
+                            conc_chl=0.8,
+                            conc_tsm=4.2,
+                        ))
+        cube.conc_chl.attrs.update({
+            "long_name": "Chlorophyll concentration",
+            "units": "mg m^-3"
+        })
+        cube.conc_tsm.attrs.update({
+            "long_name": "Total suspended matter dry weight concentration",
+            "units": "g m^-3",
+        })
+        cube.attrs.update({
+            "Conventions": "CF-1.7",
+            "coordinates": "time lat lon"
+        })
+        cube.to_zarr(cls.CUBE_PATH)
+
+    def tearDown(self) -> None:
+        rimraf(self.CUBE_PATH,
+               self.METADATA_PATH,
+               self.OPTIONS_PATH)
+
+
+class MetadataTest(unittest.TestCase):
+    def test_parse_metadata_ok(self):
+        metadata = parse_metadata(
+            {
+                "zarr_consolidated_format": 1,
+                "metadata": {
+                    ".zattrs": {
+                        "title": "Humpty-Dumpty"
+                    }
+                }
+            }
+        )
+        self.assertEqual(
+            {
+                ".zattrs": {
+                    "title": "Humpty-Dumpty"
+                }
+            },
+            metadata
+        )
+
+    def test_parse_metadata_fails(self):
+        with self.assertRaises(click.ClickException) as cm:
+            parse_metadata([])
+        self.assertEqual('Invalid consolidated metadata format',
+                         f'{cm.exception}')
+
+        with self.assertRaises(click.ClickException) as cm:
+            parse_metadata({
+                "zarr_consolidated_format": 3,
+                "metadata": {}
+            })
+        self.assertEqual('Unsupported consolidated metadata version',
+                         f'{cm.exception}')
+
+        with self.assertRaises(click.ClickException) as cm:
+            parse_metadata({
+                "zarr_consolidated_format": 1,
+                "metadata": []
+            })
+        self.assertEqual('Invalid metadata format',
+                         f'{cm.exception}')
+
+        with self.assertRaises(click.ClickException) as cm:
+            parse_metadata({
+                "zarr_consolidated_format": 1,
+                "metadata": {
+                    ".zattrs": []
+                }
+            })
+        self.assertEqual('Invalid metadata format:'
+                         ' entry ".zattrs" is not an object',
+                         f'{cm.exception}')
+
+        with self.assertRaises(click.ClickException) as cm:
+            parse_metadata({
+                "zarr_consolidated_format": 1,
+                "metadata": {
+                }
+            })
+        self.assertEqual('No metadata provided',
+                         f'{cm.exception}')
+
+    def test_parse_metadata_warn(self):
```

Comment:
Should this test be checking explicitly for the warning, or is it enough that the operation succeeds for the valid part of the patch?

Reply:
That would be good. If you know how this is best done, pls post a PR :)

Reply:
You can wrap the call to `parse_metadata` in:

```
        with self.assertWarnsRegex(
                Warning,
                expected_regex='^Ignoring metadata entry "i_am_invalid_attrs": '
                               'can only patch "[*]/[.]zattrs"$'):
```